### PR TITLE
change predicate for generating owner side's attribute in 1:n relatio…

### LINKED ISSUE #19780: Bidirectional One-To-Many relationships not generating
##### **Overview of the issue**

Hello,
I have a JDL file containing a One-To-Many relationship (Using this basic example: https://www.jhipster.tech/jdl/relationships#basic-example) between two Entites Foo and Bar. You can find the JDL file below. 

The resulting interfaces are as follows:

`export interface IBar {
  id: number;
  foo?: Pick<IFoo, 'id'> | null;
}`

`export interface IFoo {
  id: number;
}`

The Foo interface is missing the `bar?: IBar[]` property.

##### **Reproduce the error**
Create a 1:* or *:1 relationship in a JDL file then run `jhipster jdl ./jdl-file.jdl`

##### **JHipster Version(s)**
JHipster v7.9.3

##### **JHipster configuration**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "oauth2",
    "baseName": "test",
    "blueprints": [],
    "buildTool": "gradle",
    "cacheProvider": "ehcache",
    "clientFramework": "angularX",
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1663589307095,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "devServerPort": 4200,
    "dtoSuffix": "DTO",
    "enableGradleEnterprise": false,
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "entities": ["Foo", "Bar"],
    "entitySuffix": "",
    "jhiPrefix": "jhi",
    "jhipsterVersion": "7.9.3",
    "languages": ["en"],
    "lastLiquibaseTimestamp": 1663590192000,
    "messageBroker": false,
    "microfrontend": false,
    "microfrontends": [],
    "nativeLanguage": "en",
    "otherModules": [],
    "packageName": "com.mycompany.myapp",
    "pages": [],
    "prodDatabaseType": "postgresql",
    "reactive": false,
    "searchEngine": false,
    "serverPort": "8080",
    "serverSideOptions": [],
    "serviceDiscoveryType": "no",
    "skipCheckLengthOfIdentifier": false,
    "skipClient": false,
    "skipFakeData": false,
    "skipUserManagement": true,
    "testFrameworks": [],
    "websocket": false,
    "withAdminUi": true
  }
}

</pre>
</details>

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Foo
entity Bar
relationship OneToMany {
  Foo{bar} to Bar{foo}
}

</pre>
</details>

##### **Environment and Tools**

openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)

git version 2.25.1

node: v16.15.0

npm: 8.5.5

Docker version 20.10.17, build 100c701

Docker Compose version v2.10.2

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>Foo.json</summary>
<pre>

{
  "applications": "*",
  "changelogDate": "20220919122311",
  "dto": "no",
  "embedded": false,
  "entityTableName": "foo",
  "fields": [],
  "fluentMethods": true,
  "jpaMetamodelFiltering": false,
  "name": "Foo",
  "pagination": "no",
  "readOnly": false,
  "relationships": [
    {
      "otherEntityName": "bar",
      "otherEntityRelationshipName": "foo",
      "relationshipName": "bar",
      "relationshipType": "one-to-many"
    }
  ],
  "service": "no"
}
</pre>
</details>

<details>
<summary>Bar.json</summary>
<pre>

{
  "applications": "*",
  "changelogDate": "20220919122312",
  "dto": "no",
  "embedded": false,
  "entityTableName": "bar",
  "fields": [],
  "fluentMethods": true,
  "jpaMetamodelFiltering": false,
  "name": "Bar",
  "pagination": "no",
  "readOnly": false,
  "relationships": [
    {
      "otherEntityName": "foo",
      "otherEntityRelationshipName": "bar",
      "relationshipName": "foo",
      "relationshipType": "many-to-one"
    }
  ],
  "service": "no"
}
</pre>
</details>

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by RamiSouai**
change predicate for generating owner side's attribute in 1:n relationships in angular

Fix #19780

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export function convert(jdlRelationships: JDLRelationship[] = [], entityNames: s
   convertedRelationships = new Map(entityNames.map(entityName => [entityName, []]));
   const relatedRelationships = getRelatedRelationships(jdlRelationships, entityNames);
   relatedRelationships.forEach((relatedRelationship, currentEntityName) => {
+    console.log(convertedRelationships);
```
Comment (Weltraumschaf) [inline jdl/converters/jdl-to-json/jdl-to-json-relationship-converter.ts:54]: @RamiSouai I think this should be removed.

File: generators/angular/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.model.ts.ejs [hunk 2]
```diff
@@ -18,11 +18,18 @@
 -%>
 <%
 const enumImports = this.generateEntityClientEnumImports(fields);
+const predicate = (relationship) => {
```
Comment (Weltraumschaf) [inline generators/angular/templates/src/main/webapp/app/entities/_entityFolder/_entityFile.model.ts.ejs:21]: @RamiSouai Maybe we should give the predicate better name to clarify what it filters.

**Comment by DanielFran** [comment]
@RamiSouai Can you review the comments?
You need also to sign the CLA and fix the formatting issues

**Comment by DanielFran** [comment]
@RamiSouai the algorithm to calculate owner side attribute has been rebuild by @mshima on main branch.
Can you please confirm issue is fixed and we can close this PR?

**Comment by RamiSouai** [comment]
> In any case this PR will not fix the problem since ownerSide is currently used in the entire generation process.

@mshima Do you mean the problem described in #19780, or something else?

**Comment by Weltraumschaf** [comment]
Can you please give us some feedback? We are stuck on the old version of JHipster and need to update.

**Comment by mshima** [comment]
`ownerSide` means the relationship column is written in that side.
Setting ownerSide at OneToMany side is wrong.

**Comment by Weltraumschaf** [comment]
What I understand: We fixed the bug in a way that it works for us by accident, but we do it the wrong way because we do not fully understand the underlying concepts. Ok, fair enough. 

Could you please help us with our issue that the generated application code breaks after upgrade? For security reasons it is quite important for us to update JHipster regularly, but we are stuck now since months, and we have no glue what to do next. We are willing to contribute to your project. But we need a little bit more help and guidance.

**Comment by mshima** [comment]
@Weltraumschaf we should replace relationship's `ownerSide` with more meaningful properties like `restApiReadOnly` and maybe `restApiOmit`. `ownerSide` means that the relationship is writable, but the should change based on the database type which logic should be reviewed.

**Comment by rseedorff** [comment]
relates to #22481

**Comment by DanielFran** [comment]
@RamiSouai, according to this [comment,](https://github.com/jhipster/generator-jhipster/issues/19780#issuecomment-1768346320) can you confirm the issue has been fixed and we can close this PR?

**Comment by Weltraumschaf** [comment]
@DanielFran yes, you can close this PR. I messaged @RamiSouai and asked him to close the PR, but he's quite busy.

**Pull request closed without merging**